### ISSUE TITLE
Importing HTS token from HashScan into MetaMask does not associate the token - works as intended, but it's a bottleneck in usability

### ISSUE THREAD

**Issue by ed-marquez** [hashgraph/hedera-mirror-node-explorer#458]
### Description

@ericleponner @simon See the following HTS fungible token: 
![image](https://user-images.githubusercontent.com/72571340/220408620-8bf0bad1-1237-4a69-8b07-66aaefc072c6.png)

HashScan provides the ability import that token into MetaMask. Doing so shows the token in MetaMask:
![image](https://user-images.githubusercontent.com/72571340/220409123-9138f528-e1d3-4d2b-8172-95a2de128a21.png)
 
Sending units of that token to the MetaMask wallet results in the following error (no association):
![image](https://user-images.githubusercontent.com/72571340/220409312-c0dad8b1-899e-404f-a598-7041b803b4a8.png)

It's vital that we i) **check if the token is already associated with the account and ii) perform the association if not.** 

For retail users, no way currently exists to associate HTS tokens with their MetaMask wallets (accounts). If we're providing a path to import those tokens in MetaMask via HashScan, it is a natural step that we also perform that association tx for them. Otherwise, it's not an optimal user experience that retail users will consider buggy - or they'll simply think Hedera tokens are difficult to work with / don't work in MetaMask.

### Steps to reproduce

See steps above.

### Additional context

This is relevant in relation to the release of Hashio for retail users in addition to developers.

### Hedera network

mainnet, testnet, previewnet

### Version

latest

### Operating system

Windows

**Comment by shemnon** [comment]
You can call the HTS System contract to associate.  However, we will need a dApp page to do that as it involves signing and sending a transaction.

**Comment by Nana-EC** [comment]
This is an interesting idea.
Danno is correct we'd need the explorer to create and provide a transaction that calls the system HTS `associateToken(address account, address token)` that the user would sign in their wallet.
In this was the explorer would serve as a DApp.

I think there's some additional downstream work needed though.
Namely system contracts represented the precompile function address need to be created  - https://github.com/hashgraph/hedera-services/issues/4126.
The right way to do this feature would be to allow users to call HTS precompile directly and not require an intermediate contract. As such we'd need this issue to be done 1st

**Comment by ed-marquez** [comment]
Hi @Nana-EC and @shemnon - I was able to do this in a single step by just adding a direct `call` to the hts system contract in the constructor of an intermediary contract. The two-step process (deployment + association) feels invisible to the end user (probably more expensive than being able to just call system contracts directly without intermediary contracts).

Here's the contract code JIC:
```
  // SPDX-License-Identifier: Apache-2.0
pragma solidity >=0.5.0 <0.9.0;
pragma experimental ABIEncoderV2;

import "https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol";
  
  contract AssociateToken {
    address htsPrecompiles = address(0x167);

    constructor(address _tokenAddress) {
        address _receiver = msg.sender;
        htsPrecompiles.call(abi.encodeWithSelector(IHederaTokenService.associateToken.selector,_receiver, _tokenAddress));
    }  
  }
```

You can [try it here with MetaMask](https://gitpod.io/#https://github.com/ed-marquez/hedera-example-json-rpc-metamask-contracts/tree/token-associator) if want.

![image](https://user-images.githubusercontent.com/72571340/226757830-86d79990-be84-485b-953a-4c26377d3cc6.png)

**Comment by mshakeg** [comment]
Hi @ericleponner and @svienot now that the IHRC facade has been implemented in [v0.38.0](https://github.com/hashgraph/hedera-services/releases/tag/v0.38.0), could token associations&dissociations for EVM accounts now be added to the explorer? Metamask support will need to be added first, however for non-EVM accounts using Blade or Hashpack this can be done with `TokenAssociateTransaction`(and is likely cheaper than calling the facade).

**Comment by SimiHunjan** [comment]
The explorer will associate a token when it is imported in MetaMask if it has not been already. The feature is complete, but not live yet.
